app/src/main/java/com/example/habitsmasher/ui/dashboard/HabitItemAdapter.java

```diff
@@ -127,67 +97,11 @@ public int getItemCount() {
     public static class HabitViewHolder extends RecyclerView.ViewHolder {
         private final TextView _habitTitle;
         private final ConstraintLayout _habitRows;
-        private Button _editButton;
-        private Button _deleteButton;
 
-        // stores itself as an instance variable so it can be passed into EditHabitFragment
-        private HabitViewHolder _habitViewHolder = this;
-
-        public HabitViewHolder(@NonNull View itemView,
-                               ObservableSnapshotArray<Habit> _snapshots,
-                               String username,
-                               Context context) {
+        public HabitViewHolder(@NonNull View itemView) {
             super(itemView);
-
             _habitRows = itemView.findViewById(R.id.habit_rows);
             _habitTitle = itemView.findViewById(R.id.habit_title);
-            _editButton = itemView.findViewById(R.id.edit_button);
-            _deleteButton = itemView.findViewById(R.id.delete_button);
-
-            _editButton.setOnClickListener(new View.OnClickListener() {
-                @Override
-                public void onClick(View v) {
-                    int buttonPos = getAdapterPosition();
-                    EditHabitFragment editHabitFragment = new EditHabitFragment(buttonPos,
-                                                                                _snapshots.get(buttonPos),
-                                                                                _habitListFragment,
-                                                                                _habitViewHolder);
-                    editHabitFragment.show(_habitListFragment.getFragmentManager(), "Edit Habit");
-                }
-            });
-
-            _deleteButton.setOnClickListener(new View.OnClickListener() {
-                int buttonPos = getAdapterPosition();
-
-                @Override
-                public void onClick(View v) {
-                    int habitPosition = getAdapterPosition();
-                    Habit habitToDelete = _snapshots.get(habitPosition);
-                    try {
-                        deleteHabitEvents(username, habitToDelete);
-                    } catch(Error e) {
-                        Log.d(TAG, "No events to delete");
-                    }
-                    _habits.deleteHabit(context, username, habitToDelete, habitPosition);
-                }
-            });
-
-        }
```

Comment:
Edit and Delete button click function has been reworked with RecyclerTouchListener